Update translated strings for stream to channel rename.

laurynmm opened the pull request:
Updates user-facing, translated strings for the stream to channel rename (#28468) project.

Currently every separate commit should pass relevant test suites for the changes. I tried to structure them for ease of review. I imagine that some of them could be squashed after being reviewed and before being committed if we think that would be better.

---

**Backend changes**

Strings that needed to be updated were identified with:
```
git grep -i -e '".*stream' --and --not -e msgstr locale/ru/LC_MESSAGES/
```

<details>
<summary>git grep results</summary>

```console
$ git grep -i -e '".*stream' --and --not -e msgstr locale/ru/LC_MESSAGES/
locale/ru/LC_MESSAGES/django.po:msgid "Public streams"
locale/ru/LC_MESSAGES/django.po:msgid "Private streams"
locale/ru/LC_MESSAGES/django.po:msgid "Missing stream for chart: {chart_name}"
locale/ru/LC_MESSAGES/django.po:msgid "New streams"
locale/ru/LC_MESSAGES/django.po:"#%(stream_name)s > %(topic_name)s."
locale/ru/LC_MESSAGES/django.po:"You are receiving this because everyone was mentioned in #%(stream_name)s."
locale/ru/LC_MESSAGES/django.po:"#%(stream_name)s."
locale/ru/LC_MESSAGES/django.po:msgid "[resolved] #%(stream_name)s > %(topic_name)s"
locale/ru/LC_MESSAGES/django.po:"In Zulip, <b>streams</b> determine who gets a message. <b>Topics</b> tell "
locale/ru/LC_MESSAGES/django.po:msgid "Streams and topics in the Zulip app"
locale/ru/LC_MESSAGES/django.po:"To kick off a new conversation, just pick a stream and start a new topic. "
locale/ru/LC_MESSAGES/django.po:"<a href=\"%(move_topic_to_different_stream_link)s\">move a topic to a "
locale/ru/LC_MESSAGES/django.po:"different stream</a>."
locale/ru/LC_MESSAGES/django.po:"In Zulip, streams determine who gets a message. Topics tell you what the "
locale/ru/LC_MESSAGES/django.po:"topics (%(rename_topics_link)s), or even move a topic to a different stream "
locale/ru/LC_MESSAGES/django.po:"(%(move_topic_to_different_stream_link)s)."
locale/ru/LC_MESSAGES/django.po:msgid "Invalid default stream group name '{group_name}'"
locale/ru/LC_MESSAGES/django.po:msgid "Default stream group name too long (limit: {max_length} characters)"
locale/ru/LC_MESSAGES/django.po:"Default stream group name '{group_name}' contains NULL (0x00) characters."
locale/ru/LC_MESSAGES/django.po:msgid "Invalid default stream group {group_name}"
locale/ru/LC_MESSAGES/django.po:"'{stream_name}' is a default stream and cannot be added to '{group_name}'"
locale/ru/LC_MESSAGES/django.po:msgid "Default stream group '{group_name}' already exists"
locale/ru/LC_MESSAGES/django.po:"Stream '{stream_name}' is already present in default stream group "
locale/ru/LC_MESSAGES/django.po:"Stream '{stream_name}' is not present in default stream group '{group_name}'"
locale/ru/LC_MESSAGES/django.po:msgid "This default stream group is already named '{group_name}'"
locale/ru/LC_MESSAGES/django.po:msgid "Direct messages cannot be moved to streams."
locale/ru/LC_MESSAGES/django.po:msgid "Cannot change message content while changing stream"
locale/ru/LC_MESSAGES/django.po:msgid "The time limit for editing this message's stream has passed"
locale/ru/LC_MESSAGES/django.po:msgid "Expected exactly one stream"
locale/ru/LC_MESSAGES/django.po:msgid "Invalid data type for stream"
locale/ru/LC_MESSAGES/django.po:"Your bot {bot_identity} tried to send a message to stream ID {stream_id}, "
locale/ru/LC_MESSAGES/django.po:"but there is no stream with that ID."
locale/ru/LC_MESSAGES/django.po:"Your bot {bot_identity} tried to send a message to stream {stream_name}, but"
locale/ru/LC_MESSAGES/django.po:" that stream does not exist. Click [here]({new_stream_link}) to create it."
locale/ru/LC_MESSAGES/django.po:"Your bot {bot_identity} tried to send a message to stream {stream_name}. The"
locale/ru/LC_MESSAGES/django.po:" stream exists but does not have any subscribers."
locale/ru/LC_MESSAGES/django.po:msgid "Stream is already deactivated"
locale/ru/LC_MESSAGES/django.po:msgid "Stream is not currently deactivated"
locale/ru/LC_MESSAGES/django.po:msgid "Stream named {stream_name} already exists"
locale/ru/LC_MESSAGES/django.po:msgid "Stream {stream_name} un-archived."
locale/ru/LC_MESSAGES/django.po:"{user} changed the [access permissions](/help/stream-permissions) for this "
locale/ru/LC_MESSAGES/django.po:"stream from **{old_policy}** to **{new_policy}**."
locale/ru/LC_MESSAGES/django.po:"{user} changed the [posting permissions](/help/stream-sending-policy) for this stream:\n"
locale/ru/LC_MESSAGES/django.po:msgid "{user_name} renamed stream {old_stream_name} to {new_stream_name}."
locale/ru/LC_MESSAGES/django.po:msgid "{user} changed the description for this stream."
locale/ru/LC_MESSAGES/django.po:msgid "Messages in this stream will now be retained forever."
locale/ru/LC_MESSAGES/django.po:"{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
locale/ru/LC_MESSAGES/django.po:msgid "Cannot send to multiple streams"
locale/ru/LC_MESSAGES/django.po:msgid "Missing stream"
locale/ru/LC_MESSAGES/django.po:msgid "Must specify exactly 1 stream ID for stream messages"
locale/ru/LC_MESSAGES/django.po:msgid "Stream '{stream}' does not exist"
locale/ru/LC_MESSAGES/django.po:msgid "Stream with ID '{stream_id}' does not exist"
locale/ru/LC_MESSAGES/django.po:"You do not have permission to use stream wildcard mentions in this stream."
locale/ru/LC_MESSAGES/django.po:msgid "Catch up on a stream"
locale/ru/LC_MESSAGES/django.po:"Messages sent to a stream are seen by everyone subscribed to that stream. "
locale/ru/LC_MESSAGES/django.po:"Try clicking on one of the stream links below."
locale/ru/LC_MESSAGES/django.po:"In Zulip, streams [determine who gets a message](/help/streams-and-topics). "
locale/ru/LC_MESSAGES/django.po:msgid "[Browse and subscribe to streams](#streams/all)."
locale/ru/LC_MESSAGES/django.po:"In Zulip, topics [tell you what a message is about](/help/streams-and-"
locale/ru/LC_MESSAGES/django.po:"This is a private stream, as indicated by the lock icon next to the stream "
locale/ru/LC_MESSAGES/django.po:msgid "Private streams are only visible to stream members."
locale/ru/LC_MESSAGES/django.po:"To manage this stream, go to [Stream settings]({stream_settings_url}) and "
locale/ru/LC_MESSAGES/django.po:"click on `{initial_private_stream_name}`."
locale/ru/LC_MESSAGES/django.po:"This is a message on stream #**{default_notification_stream_name}** with the"
locale/ru/LC_MESSAGES/django.po:"You can learn more about topics at [Streams and "
locale/ru/LC_MESSAGES/django.po:"This is a message on stream #**{default_notification_stream_name}** with the"
locale/ru/LC_MESSAGES/django.po:msgid "Invalid data type for stream ID"
locale/ru/LC_MESSAGES/django.po:msgid "Only organization administrators can send to this stream."
locale/ru/LC_MESSAGES/django.po:"Only organization administrators and moderators can send to this stream."
locale/ru/LC_MESSAGES/django.po:msgid "Guests cannot send to this stream."
locale/ru/LC_MESSAGES/django.po:msgid "New members cannot send to this stream."
locale/ru/LC_MESSAGES/django.po:msgid "Not authorized to send to stream '{stream_name}'"
locale/ru/LC_MESSAGES/django.po:msgid "Please supply 'stream'."
locale/ru/LC_MESSAGES/django.po:msgid "Please choose one: 'stream' or 'stream_id'."
locale/ru/LC_MESSAGES/django.po:msgid "Invalid stream ID"
locale/ru/LC_MESSAGES/django.po:msgid "Stream name '{stream_name}' is already taken."
locale/ru/LC_MESSAGES/django.po:msgid "Invalid stream name '{stream_name}'"
locale/ru/LC_MESSAGES/django.po:msgid "A default stream cannot be private."
locale/ru/LC_MESSAGES/django.po:msgid "Stream(s) ({stream_names}) do not exist"
locale/ru/LC_MESSAGES/django.po:msgid "Web-public streams are not enabled."
locale/ru/LC_MESSAGES/django.po:msgid "Default stream group with id '{group_id}' does not exist."
locale/ru/LC_MESSAGES/django.po:msgid "Stream name can't be empty!"
locale/ru/LC_MESSAGES/django.po:msgid "Stream name too long (limit: {max_length} characters)."
locale/ru/LC_MESSAGES/django.po:msgid "Invalid character in stream name, at position {position}!"
locale/ru/LC_MESSAGES/django.po:msgid "Subscriber data is not available for this stream"
locale/ru/LC_MESSAGES/django.po:msgid "Unable to retrieve subscribers for private stream"
locale/ru/LC_MESSAGES/django.po:msgid "stream events"
locale/ru/LC_MESSAGES/django.po:msgid "All stream members can post"
locale/ru/LC_MESSAGES/django.po:msgid "Stream does not exist with id: {stream_id}. No invites were sent."
locale/ru/LC_MESSAGES/django.po:msgid "You do not have permission to subscribe other users to streams."
locale/ru/LC_MESSAGES/django.po:msgid "Invalid stream ID {stream_id}. No invites were sent."
locale/ru/LC_MESSAGES/django.po:msgid "Topic required when updating scheduled message type to stream."
locale/ru/LC_MESSAGES/django.po:msgid "Private streams cannot be made default."
locale/ru/LC_MESSAGES/django.po:msgid "Stream already has that name!"
locale/ru/LC_MESSAGES/django.po:msgid "{user_full_name} subscribed you to the stream {stream_name}."
locale/ru/LC_MESSAGES/django.po:msgid "{user_full_name} subscribed you to the following streams:"
locale/ru/LC_MESSAGES/django.po:msgid "Unable to access stream ({stream_name})."
locale/ru/LC_MESSAGES/django.po:msgid "You can only invite other Zephyr mirroring users to private streams."
locale/ru/LC_MESSAGES/django.po:msgid "{user_name} created the following streams: {stream_str}."
locale/ru/LC_MESSAGES/django.po:msgid "{user_name} created a new stream {stream_str}."
locale/ru/LC_MESSAGES/django.po:msgid "new streams"
locale/ru/LC_MESSAGES/django.po:msgid "**{policy}** stream created by {user_name}. **Description:**"
locale/ru/LC_MESSAGES/django.po:msgid "Not subscribed to stream id {stream_id}"
locale/ru/LC_MESSAGES/django.po:msgid "Missing stream_id"
locale/ru/LC_MESSAGES/django.po:msgid "User has disabled typing notifications for stream messages"
```
</details>

---

**Frontend changes**:

Currently only updates the translated strings on the frontend / web app code. 

```
git grep -i 'stream' locale/ru/translations.json
```

<details>
<summary>git grep results</summary>

```console
locale/ru/translations.json:  "<p>Stream will be announced in <b>#{new_stream_announcements_stream}</b>.</p>": "",
locale/ru/translations.json:  "<p>You are searching for messages that belong to more than one stream, which is not possible.</p>": "<p>Вы ищете сообщения, которые относятся более чем к одному каналу, а это невозможно.</p>",
locale/ru/translations.json:  "<strong>{name}</strong> <i>(guest)</i> is not subscribed to this stream. They will not be notified if you mention them.": "<strong>{name}</strong> <i>(гость)</i> не подписан/а на этот канал. Он/а не получит оповещение при упоминании.",
locale/ru/translations.json:  "<strong>{name}</strong> <i>(guest)</i> is not subscribed to this stream. They will not be notified unless you subscribe them.": "<strong>{name}</strong> <i>(гость)</i> не подписан/а на этот канал. Он/а не будет получать оповещения, пока вы его/ее не подпишете.",
locale/ru/translations.json:  "<strong>{name}</strong> is not subscribed to this stream. They will not be notified if you mention them.": "<strong>{name}</strong> не подписан/а на этот канал. Он/а не получит оповещение при упоминании.",
locale/ru/translations.json:  "<strong>{name}</strong> is not subscribed to this stream. They will not be notified unless you subscribe them.": "<strong>{name}</strong> не подписан/а на этот канал. Он/а не будет получать оповещения, пока вы его/ее не подпишете.",
locale/ru/translations.json:  "<z-link>Click here</z-link> to learn about exporting private streams and direct messages.": "<z-link>Нажмите здесь</z-link>, чтобы узнать больше об экспорте закрытых каналов и личных сообщений.",
locale/ru/translations.json:  "<z-user></z-user> will have the same properties as it did prior to deactivation, including role, owner and stream subscriptions.": "<z-user></z-user> будет иметь те же свойства, что и до отключения, в том числе роль, владельца и подписки на каналы.",
locale/ru/translations.json:  "<z-user></z-user> will have the same role, stream subscriptions, user group memberships, and other settings and permissions as they did prior to deactivation.": "<z-user></z-user> будут иметь ту же роль, подписки на каналы, членство в группах пользователей и другие настройки и разрешения, что и до отключения.",
locale/ru/translations.json:  "A stream needs to have a name": "Канал должен иметь название",
locale/ru/translations.json:  "A stream with this name already exists": "Канал с таким названием уже существует",
locale/ru/translations.json:  "Add default streams": "Добавить каналы по умолчанию",
locale/ru/translations.json:  "Add members. Use usergroup or #streamname to bulk add members.": "Добавить участников. Используйте название группы пользователей или #название_канала для массового добавления подписчиков.",
locale/ru/translations.json:  "Add stream": "Добавить канал",
locale/ru/translations.json:  "Add streams": "Добавить каналы",
locale/ru/translations.json:  "Add subscribers. Use usergroup or #streamname to bulk add subscribers.": "Добавить подписчиков. Используйте название группы пользователей или #название_канала для массового добавления подписчиков.",
locale/ru/translations.json:  "All messages including muted streams": "Все сообщения, включая заглушенные каналы",
locale/ru/translations.json:  "All streams": "Все каналы",
locale/ru/translations.json:  "Allow creating web-public streams (visible to anyone on the Internet)": "Разрешить создание открытых веб-каналов (видны любому в Интернет)",
locale/ru/translations.json:  "Already subscribed to {stream}": "Уже подписан на {stream}",
locale/ru/translations.json:  "Announce new stream in": "Анонсировать новый канал в",
locale/ru/translations.json:  "Archive stream": "Архивировать канал",
locale/ru/translations.json:  "Archiving stream <z-stream></z-stream> will immediately unsubscribe everyone. This action cannot be undone.": "Архивация канала <z-stream></z-stream> мгновенно отпишет всех его подписчиков. Это действие нельзя будет отменить.",
locale/ru/translations.json:  "Archiving this stream will also disable settings that were configured to use this stream:": "Архивация этого канала отключит настройки, сделанные для его использования:",
locale/ru/translations.json:  "Are you sure you want to create stream ''''{stream_name}'''' and subscribe {count} users to it?": "Вы уверены что хотите создать канал ''''{stream_name}'''' и подписать {count} пользователей на него?",
locale/ru/translations.json:  "Are you sure you want to send @-mention notifications to the <strong>{subscriber_count}</strong> users subscribed to #{stream_name}? If not, please edit your message to remove the <strong>@{stream_wildcard_mention}</strong> mention.": "Вы уверены, что хотите отправить @-упоминания для <strong>{subscriber_count}</strong> подписчиков канала #{stream_name}? Если нет, отредактируйте сообщение, убрав <strong>@{stream_wildcard_mention}</strong> упоминание.",
locale/ru/translations.json:  "Are you sure you want to unstar all messages in <stream-topic></stream-topic>?  This action cannot be undone.": "Вы уверены в том, что хотите снять отметку со всех сообщений в <stream-topic></stream-topic>?   Это действие нельзя будет отменить.",
locale/ru/translations.json:  "Automatically unmute topics in muted streams": "Автоматически включать темы в заглушенных каналах",
locale/ru/translations.json:  "Back to streams": "Назад к каналам",
locale/ru/translations.json:  "Because you are removing the last subscriber from a private stream, it will be automatically <z-link>archived</z-link>.": "Вы убираете последнего подписчика из закрытого канала, поэтому он будет автоматически <z-link>заархивирован</z-link>.",
locale/ru/translations.json:  "Because you are the only subscriber, this stream will be automatically <z-link>archived</z-link>.": "Поскольку вы единственный подписчик, этот канал будет автоматически <z-link>заархивирован</z-link>.",
locale/ru/translations.json:  "Browse 1 more stream": "Просмотреть еще 1 канал",
locale/ru/translations.json:  "Browse streams": "Просмотреть каналы",
locale/ru/translations.json:  "Browse {can_subscribe_stream_count} more streams": "Просмотр каналов - еще {can_subscribe_stream_count} шт.",
locale/ru/translations.json:  "Cannot subscribe to <z-stream></z-stream>": "Не удалось подписаться на <z-stream></z-stream>",
locale/ru/translations.json:  "Cannot subscribe to private stream <z-stream></z-stream>": "Не удалось подписаться на закрытый канал <z-stream></z-stream>",
locale/ru/translations.json:  "Cannot view stream": "Не удается просмотреть канал",
locale/ru/translations.json:  "Configure how Zulip notifies you about new messages. In muted streams, stream notification settings apply only to unmuted topics.": "Настройте оповещения Zulip о новых сообщениях. Для заглушенных каналов настройки оповещения относятся только к включенным темам.",
locale/ru/translations.json:  "Configure the default streams new users are subscribed to when joining your organization.": "Укажите каналы по умолчанию, на которые автоматически будут подписаны новые пользователи вашей организации.",
locale/ru/translations.json:  "Consider <z-link>searching all public streams</z-link>.": "Попробуйте <z-link>искать во всех открытых каналах</z-link>.",
locale/ru/translations.json:  "Create a stream": "Создать канал",
locale/ru/translations.json:  "Create new stream": "Создать новый канал",
locale/ru/translations.json:  "Create stream": "Создать канал",
locale/ru/translations.json:  "Creating stream...": "Создаю канал...",
locale/ru/translations.json:  "Currently viewing the entire stream.": "В данный момент отображается весь канал.",
locale/ru/translations.json:  "Cycle between stream narrows": "Переключаться между каналами",
locale/ru/translations.json:  "Default for stream": "По умолчанию для канала",
locale/ru/translations.json:  "Default streams": "Каналы по умолчанию",
locale/ru/translations.json:  "Default streams for new users cannot be made private.": "Каналы по умолчанию для новых пользователей нельзя делать закрытыми.",
locale/ru/translations.json:  "Default streams for this organization": "Канал по умолчанию для этой организации.",
locale/ru/translations.json:  "Demote inactive streams": "Убирать неактивные каналы",
locale/ru/translations.json:  "Edit #{stream_name}": "Редактировать #{stream_name}",
locale/ru/translations.json:  "Edit stream name and description": "Редактировать название и описание канала",
locale/ru/translations.json:  "Error creating stream": "Ошибка создания канала",
locale/ru/translations.json:  "Error in unsubscribing from #{stream_name}": "Ошибка при отписке от #{stream_name}",
locale/ru/translations.json:  "Error removing user from #{stream_name}": "Ошибка при удалении пользователя из #{stream_name}",
locale/ru/translations.json:  "Error removing user from this stream.": "Ошибка удаления пользователя с канала.",
locale/ru/translations.json:  "Exports all users, settings, and all data visible in public streams.": "Выгрузка всех пользователей, настроек, и всех данных, доступных в публичных каналах.",
locale/ru/translations.json:  "Failed adding one or more streams.": "Не удалось добавить одну или несколько тем.",
locale/ru/translations.json:  "Filter default streams": "Фильтр каналов",
locale/ru/translations.json:  "Filter streams": "Фильтр каналов",
locale/ru/translations.json:  "First time? Read our <z-link>guidelines</z-link> for creating and naming streams.": "В первый раз? Прочтите наши <z-link>рекомендации</z-link> по созданию и именованию каналов.",
locale/ru/translations.json:  "Forked from upstream at {zulip_merge_base}": "Форк апстрима с {zulip_merge_base}",
locale/ru/translations.json:  "Generate stream email address": "Создать адрес электронной почты канала",
locale/ru/translations.json:  "Go to stream settings": "К настройкам канала",
locale/ru/translations.json:  "However, it will no longer be subscribed to the private streams that you are not subscribed to.": "Тем не менее, будет прекращена подписка на закрытые каналы, на которые вы не подписаны.",
locale/ru/translations.json:  "In muted streams, stream notification settings apply only to unmuted topics.": "Для заглушенных каналов настройки оповещения относятся только к включенным темам.",
locale/ru/translations.json:  "In this stream": "",
locale/ru/translations.json:  "Includes muted streams and topics": "Включает заглушенные каналы и темы",
locale/ru/translations.json:  "Invalid stream ID": "Неверный код канала",
locale/ru/translations.json:  "Let recipients see when I'm typing messages in streams": "Показывать получателям, что я печатаю сообщение для канала",
locale/ru/translations.json:  "Let recipients see when a user is typing stream messages": "Показывать получателям, что пользователь печатает сообщение для канала",
locale/ru/translations.json:  "Log in to browse more streams": "Войдите, чтобы видеть больше каналов",
locale/ru/translations.json:  "Message #{stream_name}": "Написать в #{stream_name}",
locale/ru/translations.json:  "Message #{stream_name} > {topic_name}": "Написать в #{stream_name} > {topic_name}",
locale/ru/translations.json:  "Messages in all public streams": "Сообщения во всех открытых каналах",
locale/ru/translations.json:  "Mute stream": "Заглушить канал",
locale/ru/translations.json:  "Narrow to messages on stream <z-value></z-value>.": "Показать только сообщения канала <z-value></z-value>.",
locale/ru/translations.json:  "Narrow to stream from topic view": "Показать канал из просмотра темы",
locale/ru/translations.json:  "New stream announcements": "Объявления о новых каналах",
locale/ru/translations.json:  "New stream message": "Новое сообщение",
locale/ru/translations.json:  "New stream notifications": "Оповещения о новых каналах",
locale/ru/translations.json:  "No default streams match your current filter.": "Нет каналов по умолчанию, соответствующих текущему фильтру.",
locale/ru/translations.json:  "No matching streams": "Нет подходящих каналов",
locale/ru/translations.json:  "No stream subscribers match your current filter.": "Нет подписчиков канала, соответствующих текущему фильтру.",
locale/ru/translations.json:  "No stream subscriptions.": "Нет подписок на каналы.",
locale/ru/translations.json:  "No streams": "Никакие каналы",
locale/ru/translations.json:  "Notify stream": "Оповестить канал",
locale/ru/translations.json:  "Now following <z-link>{stream_topic}</z-link>.": "Тема <z-link>{stream_topic}</z-link> будет отслеживаться.",
locale/ru/translations.json:  "Once you leave this stream, you will not be able to rejoin.": "Если вы покините этот канала, вы не сможете вновь к нему присоединиться.",
locale/ru/translations.json:  "Only stream members can add users to a private stream.": "Только участники канала могут добавлять пользователей к закрытому каналу.",
locale/ru/translations.json:  "Only subscribers can access or join private streams, so you will lose access to this stream if you convert it to a private stream while not subscribed to it.": "Только подписчики могут иметь доступ или присоединяться к закрытым каналам, поэтому вы потеряете доступ к этому каналу, если преобразуете его в закрытый не будучи подписанными на него.",
locale/ru/translations.json:  "Only subscribers to this stream can edit stream permissions.": "Только подписчики данного канала могут изменять его разрешения.",
locale/ru/translations.json:  "Organization administrators can change the announcement stream in the organization settings.": "Администраторы организации могут изменить канал для объявлений в настройках организации.",
locale/ru/translations.json:  "Pin stream to top": "Закрепить канал сверху",
locale/ru/translations.json:  "Pin stream to top of left sidebar": "Закрепить наверху в списке каналов",
locale/ru/translations.json:  "Please specify a stream.": "Пожалуйста, укажите канал.",
locale/ru/translations.json:  "Private streams cannot be default streams for new users.": "Закрытые каналы не могут быть каналами по умолчанию для новых пользователей. ",
locale/ru/translations.json:  "Receives new stream announcements": "Получает объявления о новых каналах",
locale/ru/translations.json:  "Require topics in stream messages": "Требовать тему в сообщениях канала",
locale/ru/translations.json:  "STREAMS": "КАНАЛЫ",
locale/ru/translations.json:  "Scroll through streams": "Прокрутка каналов",
locale/ru/translations.json:  "Search all public streams in the organization.": "Искать во всех публичных каналах организации.",
locale/ru/translations.json:  "Select a stream": "Выберите канал",
locale/ru/translations.json:  "Select a stream below or change topic name.": "Выберите ниже канал или измените название темы.",
locale/ru/translations.json:  "Select a stream to subscribe": "Выберите канал для подписки",
locale/ru/translations.json:  "Select stream": "Выберите канал",
locale/ru/translations.json:  "Stream": "Канал",
locale/ru/translations.json:  "Stream color": "Цвет канала",
locale/ru/translations.json:  "Stream created recently": "Недавно созданные каналы",
locale/ru/translations.json:  "Stream creation": "Создание канала",
locale/ru/translations.json:  "Stream description": "Описание канала",
locale/ru/translations.json:  "Stream email address:": "Адрес электронной почты канала:",
locale/ru/translations.json:  "Stream name": "Имя канала",
locale/ru/translations.json:  "Stream permissions": "Разрешения канала",
locale/ru/translations.json:  "Stream settings": "Настройки канала",
locale/ru/translations.json:  "Stream successfully created!": "Канал успешно создан!",
locale/ru/translations.json:  "Streams": "Каналы",
locale/ru/translations.json:  "Streams they should join": "Каналы, к которым они должны присоединиться",
locale/ru/translations.json:  "Subscribe to <z-stream></z-stream>": "Подписаться на <z-stream></z-stream>",
locale/ru/translations.json:  "Subscribe to/unsubscribe from selected stream": "Подписаться/отписаться от выбранного канала",
locale/ru/translations.json:  "Subscribe {full_name} to streams": "Подписка {full_name} на каналы",
locale/ru/translations.json:  "Subscribed streams": "Подписка на каналы",
locale/ru/translations.json:  "The stream <b>#{stream_name}</b> does not exist. Manage your subscriptions <z-link>on your Streams page</z-link>.": "Канал <b>#{stream_name}</b> отсутствует. Управляйте подписками <z-link>на странице ваших каналов</z-link>.",
locale/ru/translations.json:  "The stream description cannot contain newline characters.": "Описание канала не должна содержать перенос строки.",
locale/ru/translations.json:  "The topic <strong>{topic_name}</strong> already exists in this stream. Are you sure you want to combine messages from these topics? This cannot be undone.": "",
locale/ru/translations.json:  "There are no default streams.": "",
locale/ru/translations.json:  "There are no streams you can view in this organization.": "В этой организации нет каналов, которые вы можете просматривать.",
locale/ru/translations.json:  "This change will make this stream's entire message history accessible according to the new configuration.": "",
locale/ru/translations.json:  "This stream does not exist or is private.": "Этот канал не существует или является закрытым.",
locale/ru/translations.json:  "This stream does not yet have a description.": "У этого канала еще нет описания.",
locale/ru/translations.json:  "This stream has been deactivated": "Этот канал был отключен",
locale/ru/translations.json:  "This stream has no subscribers.": "У этого канала нет подписчиков.",
locale/ru/translations.json:  "This stream has {sub_count, plural, =0 {no subscribers} one {# subscriber} other {# subscribers}}.": "У этого канала {sub_count, plural, =0 {нет подписчиков} one {# подписчик} other {# подписчиков}}.",
locale/ru/translations.json:  "Time limit for moving messages between streams": "Ограничение времени для перемещения сообщений в другой канал",
locale/ru/translations.json:  "Unknown stream": "Неизвестный канал",
locale/ru/translations.json:  "Unknown stream #{search_text}": "Неизвестный канал #{search_text}",
locale/ru/translations.json:  "Unmute stream": "Включить канал",
locale/ru/translations.json:  "Unmuted <z-link>{stream_topic}</z-link>.": "Включена тема <z-link>{stream_topic}</z-link>.",
locale/ru/translations.json:  "Unmuted streams and topics": "Включенные каналы и темы",
locale/ru/translations.json:  "Unpin stream from top": "Открепить канал сверху",
locale/ru/translations.json:  "Unsubscribe from <z-stream></z-stream>": "Отписаться от <z-stream></z-stream>",
locale/ru/translations.json:  "Use stream settings to unsubscribe from private streams.": "Используйте настройки канала, чтобы отписаться от закрытых каналов.",
locale/ru/translations.json:  "Use stream settings to unsubscribe the last user from a private stream.": "Используйте настройки канала, чтобы отписать последнего пользователя от закрытого канала.",
locale/ru/translations.json:  "View all streams": "Просмотреть все каналы",
locale/ru/translations.json:  "View stream": "Просмотр канала",
locale/ru/translations.json:  "View stream messages": "Показать сообщения канала",
locale/ru/translations.json:  "View streams": "Просмотр каналов",
locale/ru/translations.json:  "Warning: <strong>#{stream_name}</strong> is a private stream.": "Внимание: <strong>#{stream_name}</strong> — это закрытый канал.",
locale/ru/translations.json:  "Which parts of the email should be included in the Zulip message sent to this stream?": "Какие части электронной почты следует включать в сообщения Zulip, отправляемые в канал?",
locale/ru/translations.json:  "Who can access the stream?": "Кто имеет доступ к каналу?",
locale/ru/translations.json:  "Who can add users to streams": "Кто имеет право добавлять пользователей к каналам",
locale/ru/translations.json:  "Who can create private streams": "Кто может создавать закрытые каналы",
locale/ru/translations.json:  "Who can create public streams": "Кто может создавать открытые каналы",
locale/ru/translations.json:  "Who can create web-public streams": "Кто может создавать открытые веб-каналы",
locale/ru/translations.json:  "Who can move messages to another stream": "Кто может перемещать сообщения в другой канал",
locale/ru/translations.json:  "Who can post to the stream?": "Кто может писать сообщения в канале?",
locale/ru/translations.json:  "Who can unsubscribe others from this stream?": "Кто может отписать других от данного канала?",
locale/ru/translations.json:  "You are not currently subscribed to this stream.": "Вы не подписаны на этот канал.",
locale/ru/translations.json:  "You are not subscribed to any streams.": "Вы не подписаны ни на один канал.",
locale/ru/translations.json:  "You are not subscribed to stream <z-stream-name></z-stream-name>": "Вы не подписаны на канал <z-stream-name></z-stream-name>",
locale/ru/translations.json:  "You aren't subscribed to this stream and nobody has talked about that yet!": "Вы не подписаны на этот канал и никто об этом ещё не упоминал!",
locale/ru/translations.json:  "You can use email to send messages to Zulip streams.": "Вы можете писать в каналы Zulip по электронной почте.",
locale/ru/translations.json:  "You cannot create a stream with no subscribers!": "Вы не можете создать канал без подписчиков!",
locale/ru/translations.json:  "You do not have permission to add other users to streams in this organization.": "У вас нет права добавлять других пользователей в каналы в этой организации.",
locale/ru/translations.json:  "You do not have permission to move messages to another stream in this organization.": "",
locale/ru/translations.json:  "You do not have permission to post in this stream.": "У вас нет права писать в данном канале.",
locale/ru/translations.json:  "You do not have permission to use <b>@{stream_wildcard_mention}</b> mentions in this stream.": "Вам не разрешено использовать <b>@{stream_wildcard_mention}</b> упоминания в этом канале.",
locale/ru/translations.json:  "You have muted <z-stream-topic></z-stream-topic>.": "Вы заглушили <z-stream-topic></z-stream-topic>.",
locale/ru/translations.json:  "You must be an organization administrator to create a stream without subscribing.": "Вы должны быть администратором, чтобы создавать каналы не подписываясь.",
locale/ru/translations.json:  "You subscribed to stream <z-stream-name></z-stream-name>": "Вы подписались на канал <z-stream-name></z-stream-name>",
locale/ru/translations.json:  "You unsubscribed from stream <z-stream-name></z-stream-name>": "Вы отменили подписку на канал <z-stream-name></z-stream-name>",
locale/ru/translations.json:  "You're not subscribed to this stream. You will not be notified if other users reply to your message.": "Вы не подписаны на этот канал. Вы не получите уведомление, если кто-то ответит на ваше сообщение.",
locale/ru/translations.json:  "Your message was sent to a stream you have muted.": "Ваше сообщение отправлено в канал, который вы заглушили.",
locale/ru/translations.json:  "back to streams": "назад к каналам",
```
</details>

- Results where the only instance of "stream" in the string was "<z-stream>" were not updated.
- Results where the string is no longer in the code base either led to identifying the recently updated version of the string or identified as having been removed from the codebase.

---

**To be added to these changes:**

- frontend filter/search/wildcard adjustments for channel to stream
- URL updates (frontend and backend)
- API changelog entry

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

laurynmm commented on zerver/lib/digest.py:
```diff
@@ -255,18 +255,16 @@ def gather_new_streams(
+    channels_plain = []
 
     for stream in new_streams:
-        narrow_url = base_url + encode_stream(stream.id, stream.name)
```
This was basically doing the same thing as stream_narrow_url and updating it now will makes the URL update commits cleaner.

timabbott replied:
Cool, yeah this code path is very stale.

laurynmm commented on zerver/lib/email_notifications.py:
```diff
@@ -852,7 +852,7 @@ def enqueue_welcome_emails(user: UserProfile, realm_creation: bool = False) -> N
             move_messages_link=realm_url + "/help/move-content-to-another-topic",
             rename_topics_link=realm_url + "/help/rename-a-topic",
-            move_topic_to_different_stream_link=realm_url + "/help/move-content-to-another-stream",
+            move_channels_link=realm_url + "/help/move-content-to-another-stream",
```
The updates to the link will be part of the help center updates.

laurynmm commented on zerver/lib/streams.py:
```diff
@@ -312,22 +312,22 @@ def access_stream_for_send_message(
 
     if stream_id is not None and stream is not None:
-        raise JsonableError(_("Please choose one: 'stream' or 'stream_id'."))
+        raise JsonableError(_("Please supply only one channel parameter: name or ID."))
```
The only endpoint currently using these errors is the deprecated mute topic endpoint (https://zulip.com/api/mute-topic), so I'm not super excited about creating a new string to translate for that.

The first one could maybe be simplified to "Missing parameter"? I'm not sure about the second one though.

timabbott replied:
Yeah agreed; the first one could just use the standard missing parameter exception for the `stream_id` field, since the name use case is going to be less common.

And then maybe we should figure out a common translateable exception in `zerver/lib/exceptions` for "Unsupported parameter combination: {parameters}". The Pydantic migration may offer better tooling for this, but we're generally on the hook for own strings, so maybe a quick "api design" thread for this would be prudent, and maybe just clean it up in a little side PR.

laurynmm replied:
Started a conversation here: https://chat.zulip.org/#narrow/stream/378-api-design/topic/error.20for.20unsupported.20parameter.20combination/near/1785384

laurynmm replied:
For both parameters missing, I updated this to use the `REQ` error string for a missing required parameter: `Missing '{var_name}' argument".

timabbott replied:
Great!

laurynmm commented on zerver/actions/streams.py:
```diff
@@ -1560,18 +1560,22 @@ def send_change_stream_message_retention_days_notification(
-            new_retention_period = f"{new_value} days"
-            summary_line = f"Messages in this stream will now be automatically deleted {new_value} days after they are sent."
+            old_retention_period = _("{retention_period} days").format(retention_period=old_value)
+            new_retention_period = _("{retention_period} days").format(retention_period=new_value)
```
Made these consistent so that there should only be one new string for translators. This way "day" will be translated to the user's language.

laurynmm replied:
That works for me. Updated for `number_of_days`.

laurynmm commented on zerver/lib/exceptions.py:
```diff
@@ -174,7 +174,7 @@ def __init__(self, stream: str) -> None:
     @override
     def msg_format() -> str:
-        return _("Stream '{stream}' does not exist")
+        return _("Channel '{stream}' does not exist")
```
As noted in the commit message, the variable here, and in the ID version of this error, can't be updated without impacting clients as it would change the name of the field sent in the error response.

laurynmm commented on zerver/lib/onboarding.py:
```diff
@@ -334,7 +333,7 @@ def send_initial_realm_messages(realm: Realm) -> None:
         {
             "stream": Realm.INITIAL_PRIVATE_STREAM_NAME,
-            "topic_name": "private streams",
+            "topic_name": "private channels",
```
Not marked for translation (should these be?) but noted when manually testing.

timabbott replied:
I think they may be marked for translation in the revised onboarding, but not worth working on for now, given we're replacing it wholesale.

laurynmm commented on zerver/models/realms.py:
```diff
@@ -335,7 +335,7 @@ class Realm(models.Model):  # type: ignore[django-manager-missing] # django-stub
     DEFAULT_NOTIFICATION_STREAM_NAME = "general"
     INITIAL_PRIVATE_STREAM_NAME = "core team"
-    STREAM_EVENTS_NOTIFICATION_TOPIC_NAME = gettext_lazy("stream events")
+    STREAM_EVENTS_NOTIFICATION_TOPIC_NAME = gettext_lazy("channel events")
```
Kept this as a separate commit in case we wanted to do something different here. Note that any change here will mean a new topic for these messages.

timabbott replied:
Hmm, might be best to leave this change for a separate PR, just because there's a world where we decide it's worth trying to move all the messages from that topic to "channel events", especially given that we're linking to that topic from settings now?

laurynmm replied:
Sure. But just a quick question, since we store a message's topic as a string, would we be able to update it for existing messages with a translated version of this stream name?

laurynmm replied:
This commit is still in the updated backend changes.

laurynmm replied:
This commit is still in the updated backend changes.

laurynmm commented on web/tests/i18n.test.js:
```diff
@@ -41,10 +41,10 @@ run_test("$t", () => {
-                id: "You subscribed to stream {stream}",
-                defaultMessage: "You subscribed to stream {stream}",
+                id: "You subscribed to channel {name}",
+                defaultMessage: "You subscribed to channel {name}",
```
As of commit 3e6f34780a28, this is no longer an example of `$t` in the codebase. I don't know if it particularly matters for these tests, so I just updated it to use the modern convention of channel.

timabbott replied:
Yeah, I don't think it matters.

laurynmm commented on web/src/narrow_banner.ts:
```diff
@@ -264,28 +264,11 @@ function pick_empty_narrow_banner(): NarrowBannerData {
-                            },
-                        ),
-                    };
+                    return default_banner;
```
This change was discussed in https://chat.zulip.org/#narrow/stream/9-issues/topic/banner.20for.20unsubscribed.20stream.20with.20no.20messages/near/1783023.

laurynmm commented on web/src/stream_create.js:
```diff
@@ -428,7 +428,7 @@ export function set_up_handlers() {
         if (principals.length >= 50) {
             const html_body = render_subscription_invites_warning_modal({
-                stream_name,
+                name: stream_name,
```
I've updated these template placeholder variables in a few cases. If the string clearly has "channel" in it, then I've usually changed "stream_name" to just be "name". If it's less clear, then I've changed it to be "channel_name".

timabbott replied:
I think sticking with `channel_name` might be better if it's only a few cases, just in that the JS syntax of just passing a parameter by its name is very handy.

laurynmm replied:
Done. There are only 5 or 6 of these in the `web/` code.

timabbott commented on zerver/lib/streams.py:
```diff
@@ -312,22 +312,22 @@ def access_stream_for_send_message(
     user_profile: UserProfile, stream: Stream, sub: Optional[Subscription] = None
 ) -> None:
-    error = _("Invalid stream ID")
+    error = _("Invalid channel ID")
```
A good refactor for this file might be to just make an `InvalidChannelIDError` class and raise those exceptions rather than doing `JsonableError`s. But it's not essential for this cutover project.

timabbott commented on zerver/lib/streams.py:
```diff
@@ -472,7 +472,7 @@ def check_stream_name_available(realm: Realm, name: str) -> None:
         get_stream(name, realm)
         raise JsonableError(
-            _("Stream name '{channel_name}' is already taken.").format(channel_name=name)
+            _("Channel name '{channel_name}' is already taken.").format(channel_name=name)
```
I think it is very likely worth taking the opportunity (Not necessarily this PR, but during the time before translators redo all the translations) to consider rewriting this one and the `access_stream_by_name` one to be just "Invalid channel name" or "Channel name already in use.", without echoing the actual channel name; the client knows what it requested. Maybe worth an "api design" discussion.

laurynmm replied:
It looks like `access_stream_by_name` is used when creating/updating the default stream groups with a list of stream names, so for that case it is probably useful to know what stream name failed.

For `check_stream_name_available`, it's only used in the case of updating a single existing stream's name, so we could probably adjust that one to be "Channel name already in use."

laurynmm replied:
Updated the `check_stream_name_available` error message to be "Channel name already in use."

timabbott replied:
> It looks like access_stream_by_name is used when creating/updating the default stream groups with a list of stream names, so for that case it is probably useful to know what stream name failed.

Default stream groups aren't finished as a feature but we can defer this; doesn't seem worth more attention, so marking this one as resolved.

timabbott commented on zerver/tests/test_email_mirror.py:
```diff
@@ -1174,7 +1174,7 @@ def test_receive_email_response_for_auth_failures(self) -> None:
         self.assertEqual(
             message.content,
-            "Error sending message to stream announce via message notification email reply:\nOnly organization administrators can send to this stream.",
+            "Error sending message to stream announce via message notification email reply:\nOnly organization administrators can send to this channel.",
```
`stream announce` should be updated?

laurynmm replied:
Yes, missed this as it's not marked for translation. I assume that should also be corrected.

laurynmm replied:
Updated to now be "Error sending message to channel {channel_name} via message ..." and marked for translation.

timabbott commented on zerver/actions/streams.py:
```diff
@@ -1176,7 +1176,7 @@ def send_change_stream_permission_notification(
     with override_language(stream.realm.default_language):
         notification_string = _(
             "{user} changed the [access permissions](/help/stream-permissions) "
```
Small observation: We should probably be sure to either make the help links containing `stream` be passed as parameters (ideal) or switched to "channel" before doing a push for translators to work on updating strings.

laurynmm replied:
Quick audit of the translated strings for this shows these messages and the onboarding messages (which are being revised/changed). I'll update these to be passed as parameters.

laurynmm replied:
Updated these strings to have a parameter for the help center link.

timabbott commented on zerver/lib/onboarding.py:
```diff
@@ -172,12 +172,9 @@ def select_welcome_bot_response(human_response_lower: str) -> str:
                 + "\n\n",
-                _("[Browse and subscribe to streams](#streams/all)."),
+                _("[Browse and subscribe to channels](#streams/all)."),
             ]
```
We'll possibly end up skipping this commit, since these are being wholesale replaced in https://github.com/zulip/zulip/issues/29297; I'm just not sure which will merge first.

laurynmm replied:
This commit is still included in these changes for now and I'm following the work on the issue noted above so that I can coordinate this renaming with that update.

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -12030,7 +12030,7 @@ paths:
                           {
                             "code": "BAD_REQUEST",
-                            "msg": "Stream does not exist with id: 11. No invites were sent.",
+                            "msg": "Invalid channel ID 11. No invites were sent.",
```
Sounds correct to me!

timabbott commented on zerver/views/typing.py:
```diff
@@ -38,13 +38,13 @@ def send_notification_backend(
     if recipient_type_name == "stream":
         if stream_id is None:
-            raise JsonableError(_("Missing stream_id"))
+            raise JsonableError(_("Missing channel ID"))
```
This might be able to be standardized to match what `REQ` would do if `stream_id` hadn't been passed and that wasn't a required parameter; seems like a useless specialized string.

laurynmm replied:
Sounds good. Updated to use the same `REQ` error string: `"Missing '{var_name}' argument"`.

laurynmm commented:
**Updates**:
- Reordered the commits and prefixed all frontend / web app changes with "web" in the commit message.
- Addressed review feedback. See above comments for specific notes.

laurynmm commented:
**Updates**:
- Updated recent conversations header row to show "Channel" and not "Stream"
- Updated the message formatting help overlay to use "channel" instead of "stream" in the examples. Note this is not currently translated.

**Notes**:
- The search operators help overlay is not yet updated for channel. That will be included in a separate PR to fully transition to having "channel" and "channels" as the canonical versions of these search/narrow filters.

laurynmm commented:
Failing CI does not seem related to the changes/updates to this PR.

timabbott commented:
Merged after adding an API feature level bump, thanks for all the work on this @laurynmm!
